### PR TITLE
[PKG-2524] twine 4.0.2

### PR DESCRIPTION
**Pull request by cbouss**
[PKG-2524] twine 4.0.2

Changes:
- Update to 4.0.2

https://github.com/pypa/twine/tree/4.0.2

[PKG-2524]: https://anaconda.atlassian.net/browse/PKG-2524?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by SC426 — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,61 +1,57 @@
...
-  sha256: 28460a3db6b4532bde6a5db6755cf2dce6c5020bada8a641bb2c5c7a9b1f35b8
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/twine-{{ version }}.tar.gz
+  sha256: 9e102ef5fdd5a20661eb88fad46338806c3bd32cf1db729603fe3697b1bc83c8
 
 build:
-  noarch: python
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation --ignore-installed --no-cache-dir -vv
   number: 0
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
-  skip: true  # [py<36]
```
Comment (ryanskeith) [inline recipe/meta.yaml]: We might want to keep this for documentation purposes. Though I agree it's not necessary as we aren't building for the lowest version of python 3.7.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,61 +1,57 @@
...
-  entry_points:
-    - twine = twine.__main__:main
```
Comment (ryanskeith) [inline recipe/meta.yaml:17]: I think we definitely want to expose this entry point as twine is mostly used on the command line.

**Review by ryanskeith — CHANGES_REQUESTED** [review]
LGTM, but we should still expose the entry_points to twine.

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml]: ty, addressed

File: recipe/meta.yaml (hunk 2 above)
Comment (cbouss) [reply recipe/meta.yaml:17]: ty adressed.

**Review by skupr-anaconda — APPROVED** [review]
LGTM

**Review by lorepirri — APPROVED** [review]
LGTM!

**Review by ryanskeith — APPROVED** [review]
LGTM

**Pull request merged**